File: bindings/csharp/MikanScriptAPI.cs
using System.Threading.Tasks;

namespace MikanXR
{	
	public class MikanScriptAPI
	{
		private MikanRequestManager _requestManager;
		
		public MikanScriptAPI(MikanRequestManager requestManager) 
		{
			_requestManager= requestManager;
		}

		private static readonly string k_sendScriptMessage = "invokeScriptMessageHandler";
		public Task<MikanResponse> SendScriptMessage(string mesg) // returns MikanResult
		{
			return _requestManager.SendRequestWithPayload<string>(k_sendScriptMessage, mesg);
		}
	}
}

File: src/ClientAPI/Private/MikanScriptAPI.cpp

#include "MikanScriptAPI.h"
#include "MikanRequestManager.h"

MikanScriptAPI::MikanScriptAPI(MikanRequestManager* requestManager)
	: m_requestManager(requestManager)
{
}

MikanResponseFuture MikanScriptAPI::sendScriptMessage(const std::string& mesg)
{
	return m_requestManager->sendRequestWithPayload<std::string>(k_sendScriptMessage, mesg);
}